packages/libs/src/sdk/services/Solana/programs/BaseSolanaProgram.ts

```diff
@@ -0,0 +1,133 @@
+import {
+  Commitment,
+  Connection,
+  PublicKey,
+  TransactionMessage,
+  VersionedTransaction
+} from '@solana/web3.js'
+import type { SolanaWalletAdapter } from '../types'
+import {
+  BuildTransactionRequest,
+  BuildTransactionSchema,
+  type BaseSolanaProgramConfigInternal
+} from './types'
+import { parseParams } from '../../../utils/parseParams'
+
+const isPublicKeyArray = (arr: any[]): arr is PublicKey[] =>
+  arr.every((a) => a instanceof PublicKey)
+
+/**
+ * Abstract class for initializing individual program clients.
+ */
+export class BaseSolanaProgram {
```

Comment:
Arguably could be a separate service "SolanaClient" and composable instead of inherited. Thoughts?

Reply:
Generally in favor of composition over inheritance. But then I'm also generally in favor of functional vs. class/object-oriented 😛 .
It would be nice for a solana client to be generally usable in client code without having to create an instance of a "solana program" to use it.

Reply:
^agreed. but if too big an additional lift then fine with this also